pkg/plugins/manager/signature/manifestverifier/verifier.go

```diff
@@ -111,27 +166,77 @@ func (pmv *ManifestVerifier) GetPublicKey(keyID string) (string, error) {
 	}()
 
 	if err := json.NewDecoder(resp.Body).Decode(&data); err != nil {
-		return "", err
+		return err
 	}
 
 	if len(data.Items) == 0 {
-		return "", errors.New("missing public key")
+		return errors.New("missing public key")
+	}
+
+	cachedKeys, err := pmv.kv.ListKeys(ctx)
+	if err != nil {
+		return err
 	}
 
+	shouldKeep := make(map[string]bool)
 	for _, key := range data.Items {
-		pmv.publicKeys[key.KeyID] = key
+		err = pmv.kv.Set(ctx, key.KeyID, key.PublicKey)
+		if err != nil {
+			return err
+		}
+		shouldKeep[key.KeyID] = true
+	}
+
+	// Delete keys that are no longer in the API
+	for _, key := range cachedKeys {
+		if !shouldKeep[key] {
+			err = pmv.kv.Del(ctx, key)
+			if err != nil {
+				return err
+			}
+		}
 	}
 
-	key, exist = pmv.publicKeys[keyID]
+	// Update the last updated timestamp
+	return pmv.kv.SetLastUpdated(ctx)
+}
+
+// getPublicKey loads public keys from:
+//   - The hard-coded value if the feature flag is not enabled.
+//   - A cached value from kv storage if it has been already retrieved. This cache is populated from the grafana.com API.
+func (pmv *ManifestVerifier) getPublicKey(ctx context.Context, keyID string) (string, error) {
+	if !pmv.featureEnabled() {
+		return publicKeyText, nil
+	}
+
+	pmv.lock.Lock()
+	defer pmv.lock.Unlock()
+
+	keys, err := pmv.kv.ListKeys(ctx)
```

Comment:
Nit. Future opportunity to cache this rather than fetching the keys over and over again for each plugin being loaded.